File: flutter_app/lib/models/job_card.dart
class JobCard {
  final String jobCardNo;
  final String partyName;
  final String itemName;
  final String? itemCode;
  final double orderQty;
  final double netWeight;
  final String unit;
  final String currentDepartment;
  final String status;
  final String createdAt;

  JobCard({
    required this.jobCardNo,
    required this.partyName,
    required this.itemName,
    this.itemCode,
    required this.orderQty,
    required this.netWeight,
    required this.unit,
    required this.currentDepartment,
    required this.status,
    required this.createdAt,
  });

  factory JobCard.fromJson(Map<String, dynamic> json) {
    return JobCard(
      jobCardNo: json['jobCardNo'] ?? '',
      partyName: json['partyName'] ?? '',
      itemName: json['itemName'] ?? '',
      itemCode: json['itemCode'],
      orderQty: (json['orderQty'] as num?)?.toDouble() ?? 0.0,
      netWeight: (json['netWeight'] as num?)?.toDouble() ?? 0.0,
      unit: json['unit'] ?? 'KGS',
      currentDepartment: json['currentDepartment'] ?? 'Raw Material Store',
      status: json['status'] ?? 'In Production',
      createdAt: json['createdAt'] ?? DateTime.now().toIso8601String(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'jobCardNo': jobCardNo,
      'partyName': partyName,
      'itemName': itemName,
      'itemCode': itemCode,
      'orderQty': orderQty,
      'netWeight': netWeight,
      'unit': unit,
      'currentDepartment': currentDepartment,
      'status': status,
      'createdAt': createdAt,
    };
  }
}

File: app/src/main/java/com/aistudio/pmwtracker/model/JobCard.kt
package com.aistudio.pmwtracker.model

import androidx.compose.ui.graphics.Color
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

enum class JobCardStatus(val label: String) {
    PENDING("Pending"),
    IN_PROCESS("In Process"),
    COMPLETED("Completed"),
    HOLD("On Hold"),
    CANCELLED("Cancelled"),
    REJECTED("Rejected");

    fun getColor(): Color = when (this) {
        PENDING -> Color(0xFFF97316) // Orange
        IN_PROCESS -> Color(0xFF3B82F6) // Blue
        COMPLETED -> Color(0xFF10B981) // Green
        HOLD -> Color(0xFFEAB308) // Yellow
        CANCELLED -> Color(0xFF64748B) // Slate
        REJECTED -> Color(0xFFEF4444) // Red
    }
}

enum class JobPriority(val label: String) {
    LOW("Low"),
    MEDIUM("Medium"),
    HIGH("High"),
    URGENT("Urgent");

    fun getColor(): Color = when (this) {
        LOW -> Color(0xFF64748B)
        MEDIUM -> Color(0xFF3B82F6)
        HIGH -> Color(0xFFF97316)
        URGENT -> Color(0xFFDC2626)
    }
}

@Serializable
@Entity(tableName = "job_cards")
data class JobCard(
    @PrimaryKey val jobCardNo: String,
    val orderNo: String = "",
    val poNumber: String = "",
    val partyName: String = "",
    val itemName: String = "",
    val itemCode: String = "",
    val orderQty: Double = 0.0,
    val currentQty: Double = 0.0,
    val unit: String = "KGS",
    val balanceQty: Double = 0.0,
    val currentDepartment: Department = Department.PRODUCTION,
    val status: JobCardStatus = JobCardStatus.PENDING,
    val priority: JobPriority = JobPriority.MEDIUM,
    val heatTreatmentRequired: Boolean = false,
    val operatorName: String = "",
    val wireScrapQty: Double = 0.0,
    val targetDate: String = "",
    val deliveryDate: String = "",
    val notes: String = "",
    val rackNo: String = "",
    val binLocation: String = "",
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)
